WordPress/gutenberg
# Hide drop indicator where block isn't allowed to drop.

## What?


Fixes #24174.

If a block isn't allowed to drop in a certain area, the drop indicator shouldn't show.

In addition, the draggable chip changes color and icon when hovering over an area where there are no valid drop targets.


## Testing Instructions





1. In the post or template editor, add a few block containers of different types (Group, Gallery, Navigation, List for example) and also add a few loose blocks of different types such as Paragraph and Image.
2. Try dragging single blocks and also multiple selections of different blocks into the different containers.
3. Check that drop indicator only shows if all selected blocks being dragged are allowed in the container.
4. Try dragging single and multiple blocks around at root level and check this always works well.
5. Try dragging blocks that should only exist _inside_ certain containers (List Item, Navigation Link for example) outside of their containers - no drop indicators should show.

~~Note the transparency of the draggable chip isn't yet quite in sync with the display of valid drop targets. This can be fine-tuned but it would be great to get some initial feedback on the general approach first!~~
Update: changes to draggable chip should now be in sync with showing the drop indicator.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/8096000/5add2ee7-3c32-463b-a5fc-eaea21114154

https://github.com/WordPress/gutenberg/assets/8096000/661e1475-d880-4225-a898-bfd2185d4dc3
